core/src/test/java/io/grpc/internal/ManagedChannelImplTest.java

```diff
@@ -1193,6 +1193,18 @@ public void getState_loadBalancerDoesNotSupportChannelState() {
   }
 
   @Test
+  public void getState_loadBalancerDoesNotSupportChannelStateThenChannelShutdown() {
+    createChannel(new FakeNameResolverFactory(false), NO_INTERCEPTOR);
+    assertEquals(ConnectivityState.IDLE, channel.getState(false));
+    helper.updatePicker(mockPicker);
+
+    channel.shutdown();
```

Comment:
what about we run unit tests with a `-Dtestgrpc=true` argument, and in `ChannelExecuto`r's exception handling block we check if `testgrpc` is true, if it is we `assert false`?